src/Core/Grid/Query/Builder/EmployeeQueryBuilder.php

```diff
@@ -0,0 +1,181 @@
+        $this->contextShopIds = $contextShopIds;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $searchQueryBuilder = $this->getEmployeeQueryBuilder($searchCriteria)
+            ->select('e.*, pl.name as profile_name')
+        ;
+
+        $this->searchCriteriaApplicator->applyPagination($searchCriteria, $searchQueryBuilder);
+        $this->applySorting($searchCriteria, $searchQueryBuilder);
+
+        return $searchQueryBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $countQueryBuilder = $this->getEmployeeQueryBuilder($searchCriteria)
+            ->select('COUNT(e.id_profile)')
+        ;
+
+        return $countQueryBuilder;
+    }
+
+    /**
+     * @param SearchCriteriaInterface $searchCriteria
+     *
+     * @return QueryBuilder
+     */
+    private function getEmployeeQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $sub = $this->connection->createQueryBuilder()
+            ->select(1)
+            ->from($this->dbPrefix . 'employee_shop', 'es')
+            ->where('e.id_employee = es.id_employee')
+            ->andWhere('es.id_shop IN (:context_shop_ids)')
+        ;
+
+        $qb = $this->connection->createQueryBuilder()
+            ->from($this->dbPrefix . 'employee', 'e')
+            ->leftJoin(
+                'e',
+                $this->dbPrefix . 'profile_lang',
+                'pl', 'e.id_profile = pl.id_profile AND pl.id_lang = ' . (int) $this->contextIdLang
+            )
+            ->andWhere('EXISTS (' . $sub->getSQL() . ')')
+            ->setParameter('context_shop_ids', $this->contextShopIds, Connection::PARAM_INT_ARRAY)
+        ;
+
+        $this->applyFilters($qb, $searchCriteria->getFilters());
+
+        return $qb;
+    }
+
+    /**
+     * Apply filters for Query builder.
+     *
+     * @param QueryBuilder $queryBuilder
+     * @param array $filters
+     */
+    private function applyFilters(QueryBuilder $queryBuilder, array $filters)
+    {
+        foreach ($filters as $filterName => $filterValue) {
+            if ('id_employee' === $filterName) {
+                $queryBuilder->andWhere('e.id_employee = :' . $filterName);
+                $queryBuilder->setParameter($filterName, $filterValue);
+                continue;
+            }
+
+            if ('profile' === $filterName) {
+                $queryBuilder->andWhere('pl.id_profile = :id_profile');
+                $queryBuilder->setParameter('id_profile', $filterValue);
+                continue;
+            }
+
+            if ('active' === $filterName) {
+                $queryBuilder->andWhere('e.active = :active');
+                $queryBuilder->setParameter('active', $filterValue);
+                continue;
+            }
+
+            $queryBuilder->andWhere("$filterName LIKE :$filterName");
```

Comment:
I am worried for possible SQL injections by using `"$filterName"` statement
Can we protect this field ?

Reply:
You're  write, the best way is to defined which fields we want to use here